orangecontrib/text/widgets/owcorpusviewer.py

```diff
@@ -153,20 +165,23 @@ def load_features(self):
         self.display_indices = []
         if self.corpus is not None:
             domain = self.corpus.domain
+            self.closeContext()
+
             self.search_features = list(filter_visible(chain(domain.variables, domain.metas)))
             self.display_features = list(filter_visible(chain(domain.variables, domain.metas)))
-            # FIXME: Select features based on ContextSetting
             self.search_indices = list(range(len(self.search_features)))
             self.display_indices = list(range(len(self.display_features)))
 
+            self.openContext(self.corpus)
+            self.set_selection()
```

Comment:
Should we also move this to `set_data`? IMO it doesn't fit into `load_features` method.